src/language/mixins.md

```diff
@@ -89,6 +89,50 @@ A `mixin` declaration defines a mixin. A `class` declaration defines a [class][]
 A `mixin class` declaration defines a class that is usable as both a regular class
 and a mixin, with the same name and the same type.
 
+Any restrictions that apply to classes or mixins also apply to mixin classes:
+
+- Mixins can't have `extends` or `with` clauses, so neither can a `mixin class`.
+- Classes can't have an `on` clause, so neither can a `mixin class`. 
+
+### `abstract mixin class`
+
+You can achieve similar behavior to the `on` directive for a mixin class. 
+Make the mixin class `abstract` and define the abstract methods its behavior 
+depends on:
+
+```dart
+abstract mixin class Musician {
+  // No 'on' clause, but an abstract method that other types must define if 
+  // they want to use (mix in or extend) Musician: 
+  void playInstrument(String instrumentName);
+
+  void playPiano() {
+    playInstrument('Piano');
+  }
+  void playFlute() {
+    playInstrument('Flute');
+  }
+}
+
+class Virtuoso with Musician { // Use Musician as a mixin
+  void playInstrument(String instrumentName) {
+    print('Plays the $instrumentName beautifully');
+  }  
+} 
+
+class Novice extends Musician { // Use Musician as a class
+  void playInstrument(String instrumentName) {
+    print('Plays the $instrumentName poorly');
+  }  
+} 
+```
+
+By declaring `Musician` abstract, you force any type that uses
```

Comment:
**issue:** [GSG](https://developers.google.com/style/code-in-text#grammatical-treatment-of-code-elements) doesn't want us to refer to a class or any code object as a noun. This should be written as:
```suggestion
By declaring the `Musician` mixin as abstract, you force any type that uses
```